docs/abi-spec.rst

```diff
@@ -622,12 +623,18 @@ As an example, the encoding of ``int16(-1), bytes1(0x42), uint16(0x03), string("
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^ string("Hello, world!") without a length field
 
 More specifically:
- - Each value type takes as many bytes as its range has.
- - The encoding of a struct or fixed-size array is the concatenation of the
-   encoding of its members/elements without any separator or padding.
- - Mapping members of structs are ignored as usual.
- - Dynamically-sized types like ``string``, ``bytes`` or ``uint[]`` are encoded without
-   their length field.
+ - During the encoding, everything is encoded in-place. This means that there is
+   no distinction between head and tail, as in the ABI encoding and the length
+   of an array is not encoded.
+ - The direct arguments of ``abi.encodePacked`` are concatenated without padding,
+   but padding is added as soon as an array is entered.
```

Comment:
Not sure how to interpret this sentence. Does that mean in `abi.packedEncode(1, 2, [1,2,3], 5)` the literal 5 is padded? (Disregard the invalid syntax and consider `5` is not a `uint8`.)

Reply:
would "padding is added inside arrays" be clearer?

Reply:
or perhaps "Direct arguments of ... that are value types are encoded (and concatenated) without padding" (and then the arrays follow)

Reply:
Isn't it easier to say something like "Value types are concatenated without padding, except when nested in arrays" ?